### ISSUE TITLE
Improve the error messages we return when getting from docker

### ISSUE THREAD

**Issue by Page-** [resin-io/resin-supervisor#90]
Docker sometimes returns just `server error` for api calls, it'd be good to rewrite that to make clear that it's a docker daemon error rather than an error on one of our servers.

**Comment by pcarranzav** [comment]
Yes, we need to improve error messages in general... I think @lifeeth was  going to take care of this?

**Comment by shaunmulligan** [comment]
@pcarranzav does the rearch of the supervisor plan for some improvements in error reporting

**Comment by Page-** [comment]
ping @pcarranzav ?

**Comment by pcarranzav** [comment]
Yeah, I'll add this to the rearchitecture, thanks for reminding me!

**Comment by gelbal** [comment]
User in the forum asking about this as well:
https://forums.balena.io/t/better-error-detection-reporting-of-invalid-values-in-docker-compose/6095
